DobbyTests/MockSpec.swift

```diff
@@ -20,27 +20,48 @@ class MockSpec: QuickSpec {
                 mock.verify()
             }
 
+            it("succeeds if all expectations match an interaction (unordered)") {
+                mock.expect(matches((any(), 1)))
+                mock.expect(matches((any(), matches { $0 == 2 })))
+                mock.record(0, 2)
+                mock.record(0, 1)
+                mock.verify()
+            }
+
+            it("succeeds if all expectations match an interaction (multiple equal interactions)") {
+                mock.expect(matches((any(), 1)))
+                mock.expect(matches((any(), 1)))
+                mock.record(0, 1)
+                mock.record(0, 1)
+                mock.verify()
+            }
+
             it("fails if an expectation is not matched") {
                 mock.expect(matches((any(), equals(3))))
+                var failureMessageSent = false
                 mock.verify { (message, _, _) in
+                    failureMessageSent = true
                     expect(message).to(equal("Expectation <(_, 3)> not matched"))
                 }
+                expect(failureMessageSent).to(beTrue())
```

Comment:
I've added these to make sure that a failure message is sent. Otherwise this test would pass even though we did not get a failure message.